spring-integration-core/src/main/java/org/springframework/integration/handler/advice/ExpressionEvaluatingRequestHandlerAdvice.java

```diff
@@ -36,53 +40,62 @@
  * containing the evaluation result in its payload and the {@code inputMessage} property containing
  * the original message that was sent to the endpoint.
  * The failure expression is NOT evaluated if the success expression throws an exception.
+ * <p>
+ * When expressions are not configured, but channels are, the default expression is evaluated
+ * just into a {@code payload} from the message.
  *
  * @author Gary Russell
  * @author Artem Bilan
+ *
  * @since 2.2
  *
  */
 public class ExpressionEvaluatingRequestHandlerAdvice extends AbstractRequestHandlerAdvice {
 
-	private volatile Expression  onSuccessExpression;
+	private static final Expression DEFAULT_EXPRESSION = new FunctionExpression<Message<?>>(Message::getPayload);
+
+	private final MessagingTemplate messagingTemplate = new MessagingTemplate();
 
-	private volatile MessageChannel successChannel;
+	private Expression onSuccessExpression;
 
-	private volatile String successChannelName;
+	private MessageChannel successChannel;
 
-	private volatile Expression onFailureExpression;
+	private String successChannelName;
 
-	private volatile MessageChannel failureChannel;
+	private Expression onFailureExpression;
 
-	private volatile String failureChannelName;
+	private MessageChannel failureChannel;
 
-	private final MessagingTemplate messagingTemplate = new MessagingTemplate();
+	private String failureChannelName;
 
-	private volatile boolean trapException = false;
+	private boolean trapException = false;
 
-	private volatile boolean returnFailureExpressionResult = false;
+	private boolean returnFailureExpressionResult = false;
 
-	private volatile boolean propagateOnSuccessEvaluationFailures;
+	private boolean propagateOnSuccessEvaluationFailures;
 
-	private volatile EvaluationContext evaluationContext;
+	private EvaluationContext evaluationContext;
 
 	/**
 	 * Set the expression to evaluate against the message after a successful
 	 * handler invocation.
+	 * Defaults to {@code payload}, if {@code successChannel} is configured.
 	 * @param onSuccessExpression the SpEL expression.
 	 * @since 4.3.7
 	 */
 	public void setOnSuccessExpressionString(String onSuccessExpression) {
-		this.onSuccessExpression = new SpelExpressionParser().parseExpression(onSuccessExpression);
+		setOnSuccessExpression(EXPRESSION_PARSER.parseExpression(onSuccessExpression));
 	}
 
 	/**
 	 * Set the expression to evaluate against the message after a successful
 	 * handler invocation.
+	 * Defaults to {@code payload}, if {@code successChannel} is configured.
 	 * @param onSuccessExpression the SpEL expression.
 	 * @since 5.0
 	 */
 	public void setOnSuccessExpression(Expression onSuccessExpression) {
+		Assert.notNull(onSuccessExpression, "'onSuccessExpression' must not be null");
```

Comment:
Breaking change; someone might have...

```java
    advice.setFailureExpression(...);
    advice.setSuccessExpression(null);
```

Just for self-documentation purposes.

Maybe 5.2 only? Since the work around is simple?

However, why are these asserts even needed? You retained the null checks in `doInvoke()`.